site/en/blog/cvd/index.md

```diff
@@ -0,0 +1,337 @@
+---
+title: "Simulating color vision deficiencies in the Blink Renderer"
+description: >
+  Why and how we implemented color vision deficiency simulation in DevTools and the Blink Renderer.
+layout: "layouts/blog-post.njk"
+authors:
+  - mathiasbynens
+date: 2020-11-19
+hero: 'image/dPDCek3EhZgLQPGtEG3y0fTn4v82/A4BYzXsym40Ehje2Z7sx.jpg'
+alt: >
+  DevTools blog image
+tags:
+  - devtools-engineering
+---
+
+This article describes why and how we implemented color vision deficiency simulation in DevTools and the Blink Renderer.
+
+{% Aside %}
+Note: If you prefer watching a presentation over reading articles, then enjoy the video below! If not, skip the video and read on.
+{% endAside %}
+
+
+{% YouTube id='34iDTeCNTz4' %}
+
+## Background: bad color contrast {: #background }
+
+[Low-contrast text](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) is the most common automatically-detectable accessibility issue on the web.
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/taKtfjiS0O1QBNusjScS.svg", alt="A list of common accessibility issues on the web. Low-contrast text is by far the most common issue.", width="800", height="342" %}
+
+According to [WebAIM's accessibility analysis of the top 1-million websites](https://webaim.org/projects/million/update#wcag:~:text=Low%20contrast%20text%0985.3%25%0986.1%25), over *86%* of home pages have low contrast. On average, each home page has [36 distinct instances](https://webaim.org/projects/million/#contrast:~:text=On%20average%2C%20home%20pages%20had%2036%20distinct%20instances%20of%20low%2Dcontrast%20text.) of low-contrast text.
```

Comment:
```suggestion
According to [WebAIM’s accessibility analysis of the top 1-million websites](https://webaim.org/projects/million/update#wcag:~:text=Low%20contrast%20text%0985.3%25%0986.1%25), over *86%* of home pages have low contrast. On average, each home page has [36 distinct instances](https://webaim.org/projects/million/#contrast:~:text=On%20average%2C%20home%20pages%20had%2036%20distinct%20instances%20of%20low%2Dcontrast%20text.) of low-contrast text.
```

This is non-blocking, but I’m curious. All of the typography from the original post seems to have been turned into ASCII characters. Quotes are no longer curly, etc. Was that deliberate?

Reply:
Yes, the linting prevent smart quote.

Reply:
**What?!**

@robdodson What’s the story here? This goes against [Chrome UX writing guidelines (Google-internal)](http://go/punctuation-standards), [the Google Material writing guidelines](https://carbon.googleplex.com/google-material/pages/grammar-punctuation#punctuation), and common typography standards.

I completely understand you may not want to _require_ proper typography for all content. However, linting for it and then forcing it to be removed seems counterproductive and backwards. Could we please remove the lint rule?

Reply:
@mathiasbynens we had a discussion about this a while back in team chat. The reason we've been enforcing removal is because folks often copy/paste from google docs to markdown and it often happens that docs will convert the quotes they're using in their code examples to smart quotes. This then breaks when someone tries to copy the code and paste it into their editor.

I have to admit that this is also heavily informed by a personal experience. A few years back at I/O, I was working with a woman who was trying to do a workshop following some of the code on developers.google.com. She had copied a smart quote into her editor, but the font she was using made it hard to identify the quote in her code. So we spent the entire workshop trying to debug her and didn't figure it out until the very end. She missed the majority of the workshop and left feeling disappointed. So please understand that I'm not trying to be "counterproductive and backwards" but am attempting to look out for our users.

I think markdown-it has a plugin called typographer that we could look into using which may properly remove smart quotes if they appear in code blocks and I'd be happy to open a ticket to look into that if folks feel really strongly about this.

Reply:
I've opened https://github.com/GoogleChrome/developer.chrome.com/issues/594 to track this. A short term fix would be to add `<!-- lint disable no-smart-quotes -->` to the top of your markdown files.

Reply:
> I have to admit that this is also heavily informed by a personal experience. A few years back at I/O, I was working with a woman who was trying to do a workshop following some of the code on developers.google.com. She had copied a smart quote into her editor, but the font she was using made it hard to identify the quote in her code. So we spent the entire workshop trying to debug her and didn't figure it out until the very end. She missed the majority of the workshop and left feeling disappointed. So please understand that I'm not trying to be "counterproductive and backwards" but am attempting to look out for our users.

If that is the goal, I see two problems with the current solution:

- The lint rule is insufficient. For example, it didn’t ban the (correct) use of `…` within a code snippet in this file. That `…` could have been part of e.g. JS array spread syntax and auto-replaced by an editor, just like quotes could. 
- False positives: it affects not just code spans and blocks, but the entire document. Perhaps we could make the lint rule affect only code spans and blocks? It sounds like that’s what you really want, and it would reduce the number of false positives to smart quotes etc. in code comments only.

Lint rules seem helpful, but as long as they’re incomplete and give false positives (affecting non-code prose), perhaps we could make them non-blocking? Ultimately, I would expect syntax errors in code examples to be caught during review. 

The `<!-- lint disable no-smart-quotes -->` workaround addresses my concern — glad that it’s acceptable to you.

> So please understand that I'm not trying to be "counterproductive and backwards" but am attempting to look out for our users.

Of course — I wasn’t saying you were! I was merely speaking from my own experience of seeing my work stripped from its correct use of punctuation.